pkg/machine/shim/host.go

```diff
@@ -447,6 +449,24 @@ func Start(mc *vmconfigs.MachineConfig, mp vmconfigs.VMProvider, dirs *machineDe
 		return fmt.Errorf("reload config: %w", err)
 	}
 
+	// if the machine cannot continue starting due to a signal, ensure the state
+	// reflects the machine is no longer starting
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		sig, ok := <-signalChan
+		if ok {
+			mc.Starting = false
+			logrus.Error("signal received when starting the machine: ", sig)
+
+			if err := mc.Write(); err != nil {
+				logrus.Error(err)
+			}
+
+			os.Exit(1)
+		}
+	}()
```

Comment:
should this be moved directly above `mc.Starting = true`? i.e. below the RequireExclusiveActive() check as this doesn't write the file yet

Reply:
Yep, that's a better spot. Fixed.